provide-storage/backup-storage-provider/run-a-bsp-node.md

```diff
@@ -0,0 +1,384 @@
+2. Add the following code:
+
+    ```dockerfile title="Dockerfile"
+    # Note: This Dockerfile is architecture-agnostic. Use
+    # `docker build --platform=linux/amd64` on Apple Silicon
+    # so Rosetta can emulate x86-64 binaries.
+    FROM ubuntu:22.04
+
+    # Install runtime dependencies for datahaven-node
+    RUN apt-get update && apt-get install -y \
+        libpq5 \
+        ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+    # Copy the datahaven-node binary into the image
+    COPY datahaven-node /usr/local/bin/datahaven-node
+    RUN chmod +x /usr/local/bin/datahaven-node
+
+    # Default entrypoint: just run the node
+    ENTRYPOINT ["datahaven-node"]
+    ```
+
+    This `Dockerfile` builds a minimal Ubuntu-based image containing the `datahaven-node` binary and its required runtime libraries. It installs only the necessary dependencies, copies the node executable into the container, and marks it as runnable. On macOS with Apple Silicon, the image must be built with `--platform=linux/amd64` so Docker can emulate the `x86-64` Linux environment that the precompiled DataHaven binaries expect. The default entrypoint runs `datahaven-node` directly, allowing the container to behave like a fully configured BSP node.
+
+3. Create a `.dockerignore` file
+
+4. Run the following command in the terminal to list the `data` folder within the `.dockerignore` file:
+
+    ```bash
+    echo "data" > .dockerignore
+    ```
+
+    This way the `data` folder and its contents won't get included in the docker image.
+
+5. Build the docker image with the following command:
+
+    !!! note
+        The following command has the `--platform=linux/amd64` flag which should be included for macOS users that have an Apple Silicon chip. The `datahaven-node` binary is built so it natively supports `x86_64` chips. In order for this flag to work, macOS users need to go to their Docker Desktop app and toggle on the `Use Rosetta for x86_64/amd64 emulation on Apple Silicon` setting in Settings -> General -> Virtual Machine Options -> Apple Virtualization Framework.
+
+    ```bash
+    docker build --platform=linux/amd64 -t datahaven-bsp:latest .
+    ```
+
+    The image name can be anything. In this tutorial, we'll use `datahaven-bsp:latest`. 
+    
+    You won't run the BSP this way, but if you wanted to, right now you could run a BSP in the background through a `datahaven-bsp` docker container with a command like this:
+
+    !!! note
+        Include `--platform=linux/amd64` only if using Apple Silicon.
+
+    ```bash
+    docker run -d \
+      --platform=linux/amd64 \
+      --name datahaven-bsp \
+      --restart unless-stopped \
+      -v "./data":/data \
+      datahaven-bsp:latest \
+        --provider \
+        --provider-type bsp \
+        --max-storage-capacity 10737418240 \
+        --jump-capacity 1073741824 \
+        --storage-layer rocks-db \
+        --storage-path /data
+    ```
+
+    To display this container's logs in terminal, run:
```

Comment:
Is this step 6, or still part of a really long step 5? EDIT: oh wow, it's still step 5. If it's optional stuff maybe put it in an expandable?

Reply:
if it's not optional, we're now running the node instead of building it so i'd separate it out into it's own step